domain/modelconfig/state/state.go

```diff
@@ -27,37 +27,51 @@ func NewState(factory coredatabase.TxnRunnerFactory) *State {
 	}
 }
 
-// AgentVersion returns the current models agent version. If no agent version
-// can be found an error satisfying [errors.NotFound] will be returned.
-func (st *State) AgentVersion(ctx context.Context) (string, error) {
+// GetModelAgentVersionAndStream returns the current models set agent
+// version and stream. If no agent version or stream has ben set then an
+// error satisfying [github.com/juju/juju/core/errors.NotFound] is returned.
+//
+// Note (tlm): We purposely return the raw string values for version and stream
+// here instead of turning them into concrete types. This is because they are
+// directly composed in to the model's config as string values.
+func (st *State) GetModelAgentVersionAndStream(
+	ctx context.Context,
+) (string, string, error) {
 	db, err := st.DB()
 	if err != nil {
-		return "", errors.Capture(err)
+		return "", "", errors.Capture(err)
 	}
 
-	q := `SELECT &dbAgentVersion.target_version FROM agent_version`
-
-	rval := dbAgentVersion{}
+	rval := dbAgentVersionAndStream{}
 
-	stmt, err := st.Prepare(q, rval)
+	stmt, err := st.Prepare(`
+SELECT &dbAgentVersionAndStream.*
+FROM   agent_version
+JOIN   agent_stream ON agent_version.stream_id = agent_stream.id
+`, rval)
 	if err != nil {
-		return "", errors.Capture(err)
+		return "", "", errors.Capture(err)
 	}
 
 	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
 		err := tx.Query(ctx, stmt).Get(&rval)
 		if errors.Is(err, sql.ErrNoRows) {
-			return errors.Errorf("agent version %w", coreerrors.NotFound)
+			return errors.Errorf(
```

Comment:
Here should be agent version not found.
Do we want to return a not found error if no stream is set(rval.Stream == "")?

Reply:
They are both required values on the table. We always set a default stream and always have an agent version if the user doesn't supply either of these values. We explicitly avoid ending up in situations where they cannot be set.